youtokentome/cpp/bpe.cpp

```diff
@@ -1835,19 +1838,23 @@ Status BaseEncoder::decode(const vector<vector<int>> &ids, vector<string> *sente
   return Status();
 }
 
-Status BaseEncoder::decode(const vector<int> &ids, string *sentence) const {
+Status BaseEncoder::decode(const vector<int> &ids, string *sentence, unordered_set<int> &ignore_ids) const {
   bool first_iter = true;
   for (auto id : ids) {
     string subword;
-    Status status = id_to_subword(id, &subword, true);
-    if (!status.ok()) {
-      return status;
-    }
-    *sentence += subword;
-    if (first_iter && sentence->at(0) == ' ') {
-      *sentence = sentence->substr(1);
+
+    auto got = ignore_ids.find(id);
+    if (got == ignore_ids.end()) {
```

Comment:
It can be squezed to if (ignore_ids.count(id) == 0).

Reply:
Fixed